### PR TITLE
Make color selector take full device width to match Figma

### LINKED ISSUE #17443: Make part of next color picker visible in color pickers scroll bar
Followup of https://github.com/status-im/status-mobile/pull/17405 cc @OmarBasem 

Discussed [here](https://discord.com/channels/624634427930312714/928625369542713396/1156865341092274256)

Currently part of next color picker is not visible wich makes it not obvious for user that color pickers are scrollable. 

We should make part of next color picker visible as it is shown on the designs 

**Expected result:** 

https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=2241-250931&mode=design

Onboarding for Mobile – Figma 2023-09-28 11-35-17

**Actual result:** 

photo_2023-09-28 11 07 33

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17443 

Fixes a UI issue on smaller screen and older devices where the ScrollView that renders the list of colors didn't show that it's scrollable because it had horizontal padding alongside the spacing which led #17443 
Also on Figma, The width is set to take the whole available screen width, So The padding was unnecessary.

| Figma | iOS before | iOS after  | Android before | Android after
| --- | --- | --- | --- | --- |
| image | image | image | screenshot | screenshot |

status: ready

### CODE REVIEW

**Review by Parveshdhull — COMMENTED** [review]
Hi @ibrkhalil,
Thank you for the PR.

We need to make the UI look the same as Figma by making the margin between circles flexible as per screen size.
Please checkout [this](https://discord.com/channels/624634427930312714/928625369542713396/1156875457854181393) discord conversation for more information.

**Review by OmarBasem — APPROVED** [review]
The component needs a `content-container-style` as the component usage on different screens have different paddings, but you don't have to worry about that as I have added it in that PR https://github.com/status-im/status-mobile/pull/17468

**Comment by OmarBasem** [comment]
@ibrkhalil I didn't check the issue description. It seems the issue is not addressed? The circles need to have dynamic margins based on screen width so that part of the last circle is visible

**Comment by ibrkhalil** [comment]
> @ibrkhalil I didn't check the issue description. It seems the issue is not addressed? The circles need to have dynamic margins based on screen width so that part of the last circle is visible

The spacing between them should allow it to show if we didn't add the horizontal margin.

**Comment by OmarBasem** [comment]
> > @ibrkhalil I didn't check the issue description. It seems the issue is not addressed? The circles need to have dynamic margins based on screen width so that part of the last circle is visible
> 
> The spacing between them should allow it to show if we didn't add the horizontal margin.

Are you sure it will show on all devices? Removing the margins may make the last circle show on some devices and not on others I think

**Comment by ibrkhalil** [comment]
> > > @ibrkhalil I didn't check the issue description. It seems the issue is not addressed? The circles need to have dynamic margins based on screen width so that part of the last circle is visible
> > 
> > 
> > The spacing between them should allow it to show if we didn't add the horizontal margin.
> 
> Are you sure it will show on all devices? Removing the margins may make the last circle show on some devices and not on others I think

You're right, But what really makes me want to make padding go away is that in the designs circles go edge to edge.

**Comment by pavloburykh** [comment]
Hi @ibrkhalil! It seems that current implementation does not fix the problem for all devices. Can we have some universal solution?


### ISSUE 1 The whole circle is visible instead of part of the next one (Samsung Galaxy A52)

Such implementation does not fix the problem as it is still not obvious that colour pickers are scrollable

Here is a screenshot form **Samsung Galaxy A52**

photo_2023-10-03 14 13 30

**Comment by pavloburykh** [comment]
### ISSUE 2 Colour pickers stick to the left edge instead of to be aligned with other elements on the page

Colour pickers selector starts from the very edge while should be alligned with other elements and have more space from the left side. 

**Actual result:** 

photo_2023-10-03 14 24 52

**Expected result:** 

[Figma](https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=2241-250931&mode=design&t=BPrn944pdSiyzad2-0)

Onboarding for Mobile – Figma 2023-10-03 14-24-18

**Comment by pavloburykh** [comment]
Hey @ibrkhalil any update on this?

**Comment by ibrkhalil** [comment]
Not yet, Working on Top bar animation bugs atm.

**Comment by ibrkhalil** [comment]
> @ibrkhalil I didn't check the issue description. It seems the issue is not addressed? The circles need to have dynamic margins based on screen width so that part of the last circle is visible

Unfortunately Omar, The issue still persists.
CleanShot 2023-10-21 at 11 48 12

The idea that I was pushing for is that how do we expect a variable width to contain the same amount of elements in the same way? I think width should be dynamic to have exactly the same UI on different devices with different widths.

**Comment by ibrkhalil** [comment]
I was able to come up with a ratio to calculate a dynamic margin that'll match Figma for iPhone 11 Pro and adjust the value accordingly for other screens.

CleanShot 2023-10-21 at 12 54 11

**Comment by ibrkhalil** [comment]
@pavloburykh Ready again

File: src/quo/components/colors/color/style.cljs
Comment (J-Son89) [inline src/quo/components/colors/color/style.cljs]: I think this deserves a comment - or put this in a private function to help explain the idea/logic here 👍

Comment (ibrkhalil) [reply src/quo/components/colors/color/style.cljs]: Done

**Comment by pavloburykh** [comment]
@ibrkhalil thanx for the fix! LGTM. PR is ready for design review cc @Francesca-G

**Comment by ibrkhalil** [comment]
@pavloburykh Do you think this should be added to the wallet creation screen?
I believe it uses the same color picker, I don't know if it's required there or not though.

**Comment by pavloburykh** [comment]
> @pavloburykh Do you think this should be added to the wallet creation screen? I believe it uses the same color picker, I don't know if it's required there or not though.

@ibrkhalil According to [Figma](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1-132380&mode=design&t=GhX8KCPhMIZZvPgm-0) wallet color pickers are the same. 

Wallet for Mobile – Figma 2023-10-26 11-17-16

**Currently it looks like this:**

**Android 12, Samsung Galaxy A52**

photo_2023-10-26 11 20 11

**IOS 16.1, iPhoneX**

photo_2023-10-26 11 20 33

@ibrkhalil please reach someone from wallet team on this question. I believe @J-Son89 might be the right person. 

Meanwhile, I propose not to block this PR from merging. Just create a followup after synchronising with wallet team.

File: src/status_im2/contexts/onboarding/create_profile/view.cljs
Comment (J-Son89) [inline src/status_im2/contexts/onboarding/create_profile/view.cljs]: where is this magic 18.75 coming from?

**Review by Francesca-G — APPROVED** [review]
Looks good to me ✨

**Comment by J-Son89** [comment]
@ibrkhalil, @pavloburykh this color picker is used on several screens in the wallet. If you search for quo/color-picker you will find the uses Ibrahem. Create account page, edit account page are some examples.

File: src/status_im2/contexts/onboarding/create_profile/view.cljs
Comment (ibrkhalil) [reply src/status_im2/contexts/onboarding/create_profile/view.cljs]: I wanted to add padding 20 like the design, But I wanted to make it responsive.
18.75 = iPhone 11 Pro VW / 20

**Comment by ibrkhalil** [comment]
> @ibrkhalil, @pavloburykh this color picker is used on several screens in the wallet. If you search for quo/color-picker you will find the uses Ibrahem. Create account page, edit account page are some examples.
> 
> 

I know where it's used Jamie, But my point is should I fix this bug there too or not?

File: src/status_im2/contexts/onboarding/create_profile/view.cljs
Comment (J-Son89) [reply src/status_im2/contexts/onboarding/create_profile/view.cljs]: Ok, well this should be in a def and then leave a comment explaining the value or some means to communicate this idea 👍

Comment (ibrkhalil) [reply src/status_im2/contexts/onboarding/create_profile/view.cljs]: Okay, Will do

**Comment by J-Son89** [comment]
Ah got it, we will be making adjustments to the wallet pages anyway so feel free to leave it to keep this pr scope smaller. 👍

**Comment by OmarBasem** [comment]
Hi @ibrkhalil, I am getting the following bug in the color picker in Wallet

Screenshot_20231027_074516_Status Debug

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil, I am getting the following bug in the color picker in Wallet
> 
> Screenshot_20231027_074516_Status Debug

Thanks for reporting it Omar, I created a PR to fix it here
https://github.com/status-im/status-mobile/pull/17748

**Pull request merged**
